temporal102/Exercise1.hs

```diff
@@ -0,0 +1,139 @@
+module Exercise1 where
+
+import Control.Monad (forever)
+import Control.Monad.IO.Class (liftIO)
+import Control.Monad.Logger (defaultOutput, runStdoutLoggingT)
+import DiscoverInstances (discoverInstances)
+import RequireCallStack (RequireCallStack, provideCallStack)
+import System.IO (stdout)
+import Temporal.Core.Client (connectClient, defaultClientConfig)
+import Temporal.Runtime (TelemetryOptions (..), initializeRuntime)
+import Temporal.TH (WorkflowFn, ActivityFn)
+import Temporal.TH qualified
+import Temporal.Worker (Worker, WorkerConfig, startWorker)
+import Temporal.Worker qualified as Worker
+import Temporal.Workflow qualified as Workflow
+import Temporal.Activity (Activity)
+import Temporal.Duration (seconds)
+import UnliftIO.Concurrent (threadDelay)
+import UnliftIO.Exception (bracket)
+import Data.Aeson
+import Data.Text (Text)
+import qualified Data.Text as T
+import qualified Data.Map as Map
+import GHC.Generics (Generic)
+
+-- | Exercise 1: Durable Execution
+-- 
+-- This exercise demonstrates Temporal's core feature: durable execution.
+-- Workflows will continue to run, even when workers are killed and restarted.
+-- 
+-- Key concepts:
+-- - Workflows survive worker failures
+-- - State is automatically persisted and recovered
+-- - No duplicate work is performed during failover
+
+-- | Input data for the hello/goodbye translation workflow
+data TranslationInput = TranslationInput
+  { name :: Text           -- Name to greet
+  , languageCode :: Text   -- Target language (fr, es, de, pt)
+  } deriving (Generic, FromJSON, ToJSON, Show)
+
+-- | Output data containing translated greetings
+data TranslationOutput = TranslationOutput
+  { helloMessage :: Text     -- Translated hello message
+  , goodbyeMessage :: Text   -- Translated goodbye message
+  } deriving (Generic, FromJSON, ToJSON, Show)
+
+-- | Translation dictionary mapping language codes to term translations
+-- Supports French (fr), Spanish (es), and German (de)
+translations :: Map.Map Text (Map.Map Text Text)
+translations = Map.fromList
+  [ ("fr", Map.fromList [("hello", "bonjour"), ("goodbye", "au revoir")])
+  , ("es", Map.fromList [("hello", "hola"), ("goodbye", "adiós")])
+  , ("de", Map.fromList [("hello", "hallo"), ("goodbye", "auf wiedersehen")])
+  , ("pt", Map.fromList [("hello", "olá"), ("goodbye", "tchau")])
+  ]
+
+-- Activities are the basic unit of work in Temporal
+-- side effects and can be retried automatically on failure
+translateActivity :: (Text, Text) -> Activity () Text
+translateActivity (term, lang) = do
+  case Map.lookup lang translations >>= Map.lookup (T.toLower term) of
+    Just translation -> return translation
+    Nothing -> return term
```

Comment:
nit: we've been using `pure` rather than `return` in general, although there should be no difference here.

shouldn't matter if the person doing these exercises is familiar with Haskell, but it could be a little misleading if one thinks there's a deeper reason for using this over the other.